shared/fs/common/path-item-action.js

```diff
@@ -0,0 +1,129 @@
+// @flow
+import * as React from 'react'
+import * as Types from '../../constants/types/fs'
+import * as Constants from '../../constants/fs'
+import {globalStyles, globalColors, globalMargins, isMobile, platformStyles} from '../../styles'
+import {Box, ClickableBox, Icon, Text} from '../../common-adapters'
+import FloatingMenu, {
+  FloatingMenuParentHOC,
+  type FloatingMenuParentProps,
+} from '../../common-adapters/floating-menu'
+import {type MenuItem} from '../../common-adapters/popup-menu'
+import PathItemIcon from '../common/path-item-icon'
+import PathItemInfo from '../common/path-item-info'
+import StaticBreadcrumb from '../common/static-breadcrumb'
+import {memoize} from 'lodash'
+
+type Props = {
+  name: string,
+  size: number,
+  type: Types.PathType,
+  lastModifiedTimestamp: number,
+  lastWriter: string,
+  childrenFolders: number,
+  childrenFiles: number,
+  itemStyles: Types.ItemStyles,
+  actionIconClassName?: string,
+  actionIconFontSize?: number,
+  pathElements: Array<string>,
+  menuItems: Array<MenuItem>,
+}
+
+const PathItemAction = (props: Props & FloatingMenuParentProps) => (
+  <Box>
+    <ClickableBox onClick={props.toggleShowingMenu} ref={props.setAttachmentRef}>
+      <Icon
+        type="iconfont-ellipsis"
+        color={globalColors.black_40}
+        style={actionIconStyle}
+        fontSize={props.actionIconFontSize}
+        className={props.actionIconClassName}
+      />
+    </ClickableBox>
+    <FloatingMenu
+      containerStyle={stylesFloatingContainer}
+      attachTo={props.attachmentRef}
+      visible={props.showingMenu}
+      onHidden={props.toggleShowingMenu}
+      position="bottom right"
+      closeOnSelect={true}
+      header={{
+        title: 'unused',
+        view: (
+          <Box style={stylesHeader}>
```

Comment:
you might want to break this out into its own component, can be in this file but outside of all this nesting